mappings/net/minecraft/item/ItemStack.mapping

```diff
@@ -66,6 +66,8 @@ CLASS net/minecraft/class_31 net/minecraft/item/ItemStack
 		ARG 2 attacker
 	METHOD method_706 writeNbt (Lnet/minecraft/class_8;)Lnet/minecraft/class_8;
 		ARG 1 nbt
+	METHOD method_707 getDamageOnEntity (Lnet/minecraft/class_57;)I
```

Comment:
Should probably be `getAttackDamageModifier`, according to `Item#ATTACK_DAMAGE_MODIFIER_ID` in yarn

Reply:
Closes to this is probably `getAttackDamage()` in `SwordItem` (an other tool classes) in yarn, however that does not take in an entity as an argument, meanwhille the beta method does. Looking at other methods like useOnEntity it seems likeadding the "onEntity" part when an entity is taken in as argument is probably a "yarny" approach.
The ATTACK_DAMAGE_MODIFIER_ID is some sort of UUID which seems to be passed into a builder and used when constructing an EntityAttribute Modifier

Reply:
I'd argue that `public float getMiningSpeedMultiplier(ItemStack stack, Block block)` doesn't have `OnBlock` suffix, so `getAttackDamageModifier` would fit that naming scheme better.

`ATTACK_DAMAGE_MODIFIER_ID` is pretty much the direct successor to this method anyway, just in the form of an item attribute.

Reply:
What about just `getAttackDamage` without the suffix then, pretty much 1:1 to the yarn method

Reply:
That method is:
1. Not in the base item class, while `ATTACK_DAMAGE_MODIFIER_ID` is
2. Used only by mobs to determine which sword/tool they prefer to pick up.

Though, if we look at `ToolMaterials` class, where both mining speed and attack damage getter methods are present, mining speed is returned by `getMiningSpeedMultiplier`, which is the same name as in Item class, while attack damage is returned by `getAttackDamage`, so it does make sense to take this name from `ToolMaterials` for parity with `getMiningSpeedMultiplier` being the same across the two classes.

So yeah, `getAttackDamage` does make sense.